src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/TaskAutoRoleAssignmentServiceTest.java

```diff
@@ -243,6 +275,97 @@ void should_assign_task_to_highest_priority_user_when_more_than_one_role_assignm
 
         assertThat(autoAssignCFTTaskResponse.getAssignee())
             .isEqualTo(roleAssignmentResource1.getActorId());
+
+        //set tribunal-caseworker high prioritised user
+        taskRoleResources = Set.of(
+            taskRoleResource("tribunal-caseworker", true, 3),
+            taskRoleResource("senior-tribunal-caseworker", true, 2)
+        );
+        taskResource.setTaskRoleResources(taskRoleResources);
+
+        when(roleAssignmentService.queryRolesForAutoAssignmentByCaseId(taskResource))
+            .thenReturn(roleAssignments);
+
+        autoAssignCFTTaskResponse = taskAutoAssignmentService.autoAssignCFTTask(taskResource);
+
+        assertEquals(CFTTaskState.ASSIGNED, autoAssignCFTTaskResponse.getState());
+
+        assertThat(autoAssignCFTTaskResponse.getAssignee())
+            .isEqualTo(roleAssignmentResource2.getActorId());
+
+    }
+
+    @Test
+    void auto_assign_should_return_task_assignee_when_auto_assignable_is_true_and_authorisation_is_null() {
+        TaskResource taskResource = createTaskResource();
+
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+
+        //first role assignment set
+        RoleAssignment roleAssignmentResource1 = createRoleAssignment(
+            "lowPrioritisedUser",
+            "tribunal-caseworker",
+            List.of("IA", "DIVORCE", "PROBATE")
+        );
+
+        roleAssignments.add(roleAssignmentResource1);
+
+        //second role assignment set
+        RoleAssignment roleAssignmentResource2 = createRoleAssignment(
+            "highPrioritisedUser",
+            "senior-tribunal-caseworker",
+            List.of("IA", "DIVORCE", "PROBATE")
+        );
+        roleAssignments.add(roleAssignmentResource2);
+
+        Set<TaskRoleResource> taskRoleResources = Set.of(
+            taskRoleResource("tribunal-caseworker", true, 3,null),
+            taskRoleResource("senior-tribunal-caseworker", false, 2,new String[]{})
```

Comment:
Bit difficult to review in a test but seems like it is testing the auto-assignable flag works ok.  2nd one is `false` and so we should be assigning it to the first.  Is that correct?